spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/KerasBaseSpec.scala

```diff
@@ -14,11 +14,28 @@
  * limitations under the License.
  */
 package com.intel.analytics.bigdl.keras
-import com.intel.analytics.bigdl.nn.abstractnn.AbstractModule
+import com.intel.analytics.bigdl.Criterion
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractCriterion, AbstractModule, Activity}
 import com.intel.analytics.bigdl.tensor.Tensor
-import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
 
-class KerasBaseSpec extends FlatSpec with BeforeAndAfter with Matchers {
+import scala.sys.process._
+
+class KerasBaseSpec extends FlatSpec with BeforeAndAfterEach with Matchers {
+
+  protected def ifskipTest(): Unit = {
+    // Skip unitest if environment is not ready
+    try {
+    Seq("python", "-c", "import keras; import tensorflow").!!
+    } catch {
+      case e: Throwable => cancel("python or keras or tensorflow is not installed", e)
+    }
+  }
+
+  override def beforeEach() {
+    ifskipTest()
```

Comment:
will this be run in each test. I remember if you call cancel before test case, the test will be marked as failure. It is marked as skipped when you call cancel in each test.

Reply:
oh, that's too bad which means we need to explicitly call this `ifskipTest` in each test.

Reply:
would change it if there's no other better option.

Reply:
verify it when run keras code?